tests/std/tests/P0896R4_P1614R2_spaceship/test.cpp

```diff
@@ -0,0 +1,193 @@
+    { // multimap
+        std::multimap<char, int> a1 = {{'a', 1}, {'b', 2}, {'a', 3}};
+        std::multimap<char, int> a2 = {{'a', 1}, {'a', 3}, {'b', 2}};
+        std::multimap<char, int> b1 = {{'z', 4}, {'y', 90}, {'z', 12}};
+        ordered_containers_test(a1, a2, b1);
+    }
+    { // set
+        std::set<int> a1;
+        a1.insert(10);
+        a1.insert(20);
+
+        std::set<int> a2;
+        a2.insert(10);
+        a2.insert(20);
+
+        std::set<int> b1;
+        b1.insert(30);
+        b1.insert(40);
+        ordered_containers_test(a1, a2, b1);
+    }
+    { // multiset
+        std::multiset<int> a1;
+        a1.insert(10);
+        a1.insert(10);
+        a1.insert(20);
+
+        std::multiset<int> a2;
+        a2.insert(10);
+        a2.insert(20);
+        a2.insert(10);
+
+        std::multiset<int> b1;
+        b1.insert(30);
+        b1.insert(40);
+        b1.insert(40);
+        ordered_containers_test(a1, a2, b1);
+    }
+    { // unordered_map
+        using stringmap = std::unordered_map<std::string, std::string>;
+        stringmap a     = {{"cat", "tabby"}, {"dog", "poodle"}, {"bear", "grizzly"}};
+        stringmap b     = {{"dog", "poodle"}, {"bear", "grizzly"}, {"cat", "tabby"}};
+        stringmap c     = {{"cat", "siamese"}, {"dog", "lab"}, {"bear", "polar"}};
+        unordered_containers_test(a, b, c);
+    }
+    { // unordered_multimap
+        using stringmap = std::unordered_multimap<std::string, std::string>;
+        stringmap a     = {{"cat", "tabby"}, {"dog", "poodle"}, {"cat", "siamese"}, {"dog", "poodle"}};
+        stringmap b     = {{"dog", "poodle"}, {"cat", "siamese"}, {"cat", "tabby"}, {"dog", "poodle"}};
+        stringmap c     = {{"cat", "siamese"}, {"dog", "lab"}, {"bear", "polar"}};
+        unordered_containers_test(a, b, c);
+    }
+    { // unordered_set
+        std::unordered_set<std::string> a = {"cat", "dog", "bear"};
+        std::unordered_set<std::string> b = {"bear", "cat", "dog"};
+        std::unordered_set<std::string> c = {"mouse", "cat", "bear", "dog"};
+        unordered_containers_test(a, b, c);
+    }
+    { // unordered_multiset
+        std::unordered_multiset<std::string> a = {"cat", "dog", "cat"};
+        std::unordered_multiset<std::string> b = {"cat", "cat", "dog"};
+        std::unordered_multiset<std::string> c = {"mouse", "cat", "bear", "dog"};
+        unordered_containers_test(a, b, c);
+    }
+    { // queue
+        std::deque<int> deq1(3, 100);
+        std::deque<int> deq2(2, 200);
+        std::queue<int> a(deq1);
+        std::queue<int> b(deq1);
+        std::queue<int> c(deq2);
+        ordered_containers_test(a, b, c);
+    }
+    { // stack
+        std::stack<int> a;
+        a.push(2);
+        a.push(2);
+        std::stack<int> b;
+        b.push(2);
+        b.push(2);
+        std::stack<int> c;
+        c.push(3);
+        c.push(3);
+        ordered_containers_test(a, b, c);
+    }
+}
+
+int main() {
+    ordering_test_cases();

```

Comment:
All of the ordering tests use elements that are strongly ordered. I don't think we need to replicate all of those tests for weakly and partially ordered element types, but it would be nice to validate the metaprogramming by `static_assert`ing that `decltype(meow<T>{} <=> meow<T>{})` is correctly `weak_ordering` and `partial_ordering` for all containers `meow` when `T{} <=> T{}` returns those types.